pkg/controllers/streaming/processor_controller_test.go

```diff
@@ -285,27 +277,27 @@ func TestReconcile(t *testing.T) {
 		},
 	}, {
 		Name: "get container fails",
+		Key:  types.NamespacedName{Namespace: testNamespace, Name: testName},
 		GivenObjects: []runtime.Object{
 			processor(testNamespace, testName, containerDelta),
 			configMap(testNamespace, processorImages, map[string]string{processorImageKey: testProcessorImage}),
 		},
-		Key: types.NamespacedName{Namespace: testNamespace, Name: testName},
 		ExpectTracks: []rtesting.TrackRequest{
 			rtesting.CreateTrackRequest("", "ConfigMap", "", processorImages).By(testNamespace, testName),
 			rtesting.CreateTrackRequest("build.projectriff.io", "Container", testNamespace, testContainer).By(testNamespace, testName),
 		},
-		GetHook:       getErrorOn(&v1alpha1.Container{}, testNamespace, testContainer, scheme),
-		ShouldErr:     true,
-		ShouldRequeue: true,
-		Verify:        rtesting.AssertErrorEqual(testError),
+		WithReactors: []rtesting.ReactionFunc{
+			rtesting.InduceFailure("get", "Container"),
```

Comment:
it seems that `getErrorOn` was a little more precise than what `InduceFailure` does here.
Is there a way to induce a failure on a Container with a particular name in a particular namespace?

Reply:
create a new reactor